EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -135,63 +142,146 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch
             //mlContext.Model.Save(trainedModel, trainingData.Schema, @"C:\Users\User\Downloads\TrainedModel.zip");
         }
 
-        private static Dictionary<string, int> GetSequenceToPSMCount(List<PeptideSpectralMatch> psms)
+        private static Dictionary<string, Dictionary<int, Tuple<double, double>>> ComputeHydrophobicityValues(List<PeptideSpectralMatch> psms, bool modified)
         {
-            Dictionary<string, int> sequenceToPsmCount = new Dictionary<string, int>();
+            SSRCalc3 calc = new SSRCalc3("SSRCalc 3.0 (300A)", SSRCalc3.Column.A300);
+            Dictionary<string, Dictionary<int, Tuple<double, double>>> rtHydrophobicityAvgDev = new Dictionary<string, Dictionary<int, Tuple<double, double>>>();
 
-            List<string> sequences = new List<string>();
-            foreach (PeptideSpectralMatch psm in psms)
-            {
-                var ss = psm.BestMatchingPeptides.Select(b => b.Peptide.FullSequence).ToList();
-                sequences.Add(String.Join("|", ss));
-            }
+            List<string> filenames = psms.Select(f => f.FullFilePath).ToList();
 
-            var s = sequences.GroupBy(i => i);
+            filenames = filenames.Distinct().ToList();
 
-            foreach (var grp in s)
+            foreach (string filename in filenames)
             {
-                sequenceToPsmCount.Add(grp.Key, grp.Count());
-            }
-            return sequenceToPsmCount;
-        }
+                Dictionary<int, List<double>> hydrobophobicites = new Dictionary<int, List<double>>();
+                Dictionary<int, Tuple<double, double>> averagesCommaStandardDeviations = new Dictionary<int, Tuple<double, double>>();
 
-        /// <summary>
-        /// This method goes throught the complete set of peptide spectral matches and counts the number of appearances for each accession number.
-        /// This includes all of the ambiguous entries. So, if one peptide spectral match has three peptides with three accessions, all are counted.
-        /// </summary>
-        /// <param name="psms"></param>
-        /// <returns></returns>
-        public static Dictionary<string, int> GetAccessionCounts(List<PeptideSpectralMatch> psms)
-        {
-            Dictionary<string, int> accessionCountDictionary = new Dictionary<string, int>();
-
-            foreach (PeptideSpectralMatch psm in psms)
-            {
-                if (psm != null)
+                foreach (PeptideSpectralMatch psm in psms.Where(f => (f.FullFilePath == filename || f.FullFilePath == null) && f.FdrInfo.QValue <= 0.01))
```

Comment:
` || f.FullFilePath == null` FullFilePath should never be null, this should be removed. i'm assuming this is for unit tests. but the unit test should be better, instead of adding stuff like this to production code to compensate for poorly constructed unit tests. again, not immediately necessary, but fixing stuff like this should be the next PR.